lib/Frontend/Frontend.cpp

```diff
@@ -640,8 +640,9 @@ bool CompilerInstance::setUpVirtualFileSystemOverlays() {
     return true;
   }
 
-  if (Invocation.getCASOptions().requireCASFS()) {
-    if (Invocation.getCASOptions().HasImmutableFileSystem) {
+  if (CASOpts.EnableCaching) {
+    if (!CASOpts.ClangIncludeTree.empty() ||
+        !CASOpts.ClangIncludeTreeFileList.empty()) {
```

Comment:
I'm not sure what is going on with this change

For `requireCASFS` changing to `EnableCaching`: what happens if we have `InputFileKey` or `BridgingHeaderPCHCacheKey` but not caching? 

For `HasImmutableFileSystem`: isn't this the same conditions as that flag? What's the reason to change it here?

Reply:
> For requireCASFS changing to EnableCaching: what happens if we have InputFileKey or BridgingHeaderPCHCacheKey but not caching?

The entire bag of inputs from CAS are tied with EnableCaching. They are basically all ignored if caching is not enabled.

HasImmutableFileSystem is actually not the same has the check here when it is inside a sub invocation. Maybe HasImmutableFileSystem is wrong and should be turned into a instance method that needs to be computed from current CASOption state.

Reply:
The code change is actually no-op, just to make it more readable. If requires cleanup, I will do in a separate commit.